### PR TITLE
feat: fix Picasso components for SSR

### PR DESCRIPTION
**Pull request by yusufzmly**
[FX-2335]

### Description

- Add isBrowser function
- Add isBrowser control for NumberInput and Tooltip component when they are using window object. Because usage of window object makes SSR fail. 

### How to test

- There shouldn't be any break or change for NumberInput and Tooltip component, It can be tested by unit tests or storybook.

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and
1. You have no extra requests.
2. You have optional requests.
   1. Add `nit: ` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because
1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:
1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2335]: https://toptal-core.atlassian.net/browse/FX-2335?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso/src/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -70,10 +70,14 @@ const useTooltipState = ({ externalOpen }: UseTooltipStateOptions) => {
...
-  const isTouchDevice = !isPointerDevice()
+const getDelayDuration = (delay: DelayType): undefined | number => {
+  const pointerDevice = isPointerDevice()
+
+  if (pointerDevice !== undefined) {
+    return pointerDevice ? delayDurations[delay] : 0
+  }
 
-  return isTouchDevice ? 0 : delayDurations[delay]
+  return undefined
```
Comment (TomasSlama) [inline packages/picasso/src/Tooltip/Tooltip.tsx:80]: do you think it would be problematic to return 0 on the server?

File: packages/picasso/src/Tooltip/styles.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export default ({ palette, shadows, sizes: { borderRadius } }: Theme) =>
       fontSize: rem('13px'),
       lineHeight: '1.5em',
       padding: '1rem',
-      borderRadius: borderRadius.small,
+      borderRadius: sizes?.borderRadius?.small,
```
Comment (TomasSlama) [inline packages/picasso/src/Tooltip/styles.ts:87]: why is this needed? is `sizes` `undefined` on server?

Comment (yusufzmly) [reply packages/picasso/src/Tooltip/styles.ts:87]: I got this undefined while i was trying in my local. But i can try again.

**Comment by TomasSlama** [comment]
CI is failing because of eslint error in NumberInput, and you need to follow commit conventions to as Danger suggests to you

**Comment by TomasSlama** [comment]
@toptal-bot run all

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

File: packages/picasso/src/NumberInput/NumberInput.tsx [hunk 3]
```diff
@@ -61,6 +64,10 @@ const NumberAdornment = (props: NumberAdornmentProps) => {
   const normalizedMax = Number(max)
 
   const fireEvent = (nextValue: number) => {
+    if (typeof nativeInputValueSetter !== 'function') {
+      return
+    }
```
Comment (dmaklygin1) [inline packages/picasso/src/NumberInput/NumberInput.tsx:69]: I think, comparing it with `undefined` is quite enough, it may be either 100% function or undefined

**Comment by yusufzmly** [comment]
@toptal-bot run package:alpha-release

File: packages/picasso/src/Tooltip/styles.ts (hunk 2 above)
Comment (TomasSlama) [reply packages/picasso/src/Tooltip/styles.ts:87]: did you check? I don't recall we would have to use like this anywhere

**Review by TomasSlama — APPROVED** [review]

File: packages/picasso/src/Tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -70,10 +70,14 @@ const useTooltipState = ({ externalOpen }: UseTooltipStateOptions) => {
   }
 }
 
-const getDelayDuration = (delay: DelayType) => {
-  const isTouchDevice = !isPointerDevice()
+const getDelayDuration = (delay: DelayType): undefined | number => {
+  const pointerDevice = isPointerDevice()
+
+  if (pointerDevice !== undefined) {
+    return pointerDevice ? delayDurations[delay] : 0
```
Comment (dmaklygin1) [inline packages/picasso/src/Tooltip/Tooltip.tsx:77]: it's quite a confusing, you've already check `pointerDevice` whether it's undefined or not, and now you check it again.
whether `isPointerDevice()` may return `null` or whathever?
Also, I think, better to specify a return type of `isPointerDevice` function to avoid confusion here. Thanks :)

Comment (yusufzmly) [reply packages/picasso/src/Tooltip/Tooltip.tsx:77]: @dmaklygin1 I will add return type but before this, I haven't checked before if is undefined or not. I check if it is onBrowser or not. And according to it, I am returning undefined or boolean. Here, I am checking if it is undefined because I dont want to return 0 here. If i return 0, it means it will be used as enterDelay and enterNextDelay props on MuiTooltip. I wanted to disable these props since we dont know if it is a pointer device or not at the server. Wdyt ?

Comment (OndrejTuma) [reply packages/picasso/src/Tooltip/Tooltip.tsx:77]: I have another suggestion, hopefully less confusing. Here you can just use `isBrowser()` and return undefined if true and revert changes in `isPointerDevice()` function, wdyt?

Comment (yusufzmly) [reply packages/picasso/src/Tooltip/Tooltip.tsx:77]: Yes i think that would be better solution. Let me see

File: packages/picasso/src/Tooltip/Tooltip.tsx [hunk 5]
```diff
@@ -70,10 +70,22 @@ const useTooltipState = ({ externalOpen }: UseTooltipStateOptions) => {
...
+
+const getDelayDuration = (delay: DelayType): number | undefined => {
+  const isTouchDevice = checkIfItIsATouchDevice()
 
-  return isTouchDevice ? 0 : delayDurations[delay]
+  return isTouchDevice === undefined
+    ? undefined
+    : isTouchDevice
+    ? 0
+    : delayDurations[delay]
```
Comment (OndrejTuma) [inline packages/picasso/src/Tooltip/Tooltip.tsx:88]: I think it's still confusing to me, there is a lot of conditions.
Correct me if I'm wrong but would this also work?
```diff
const getDelayDuration = (delay: DelayType) => {
+ if (!isBrowser()) {
+   return undefined
+ }

  const isTouchDevice = !isPointerDevice()

  return isTouchDevice ? 0 : delayDurations[delay]
}
```

**Review by OndrejTuma — APPROVED** [review]
If your tests pass, we're good to go 👍

**Comment by yusufzmly** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@toptal-bot run all
